### PR TITLE
Users components

### PR DESCRIPTION
**Pull request by aviehu**

### CODE REVIEW

File: src/components/UsersItem.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import TableRow from '@material-ui/core/TableRow';
+import TableCell from '@material-ui/core/TableCell';
+import PropTypes from 'prop-types';
+
+function UserItem(props) {
+
+    return (
```
Comment (aviehu) [inline src/components/UsersItem.js]: @roeeyud props.somthing cause an eslint error even when in the end i have validations

File: src/components/UsersItem.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        <TableRow>
+            <TableCell>{props.id}</TableCell>
+            <TableCell>{props.userName}</TableCell>
+            <TableCell>{props.age}</TableCell>
+        </TableRow>
+    )
+}
```
[the remark this answers is not in the document]
Comment (aviehu) [reply src/components/UsersItem.js]: ok cool ill try it

[the remark this answers is not in the document]
Comment (aviehu) [reply src/components/UsersItem.js]: tried it and i got the same result

File: src/stories/index.js [hunk 3]
```diff
@@ -3,10 +3,32 @@ import { storiesOf } from '@storybook/react';
...
+},
+{
+  id: '3',
+  userName: 'yossi',
+  age: '28'
+}];
 
 storiesOf('Welcome', module).add('to Storybook', () => <Welcome showApp={linkTo('Button')} />);
 
+storiesOf('User Components', module)
```
Comment (roeeyud) [inline src/stories/index.js]: Each component should have its own story

File: src/components/UserListBuilder.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    border-color: red;
+`;
+
+class UserListBuilder extends Component {
+    state = {
+        users: null,
+        formOpen: false
+    }
+
+    clickHandler() {
```
Comment (roeeyud) [inline src/components/UserListBuilder.js]:
```
clickHandler = () => {
   this.setState({ formOpen: true })
}

```

File: src/components/UserListBuilder.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+                    size={200}
+                    color={'#123abc'}
+                    loading={this.state.loading}
+                />
+            )
+        }
+        else {
+            return (
+                <div>
+                    <UserList users={this.state.users} click={this.clickHandler}></UserList>
```
Comment (roeeyud) [inline src/components/UserListBuilder.js]:
```
click={this.clickHandler.bind(this)}
```

File: src/components/UserItem.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import TableRow from '@material-ui/core/TableRow';
+import TableCell from '@material-ui/core/TableCell';
+
+function UserItem({ id, userName, age, click }) {
+    return (
+        <TableRow onClick={click}>
```
Comment (roeeyud) [inline src/components/UserItem.js]: onClick={() => click(id)}

File: src/components/UserList.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+import TableHead from '@material-ui/core/TableHead';
+import { TableRow, TableCell } from '@material-ui/core';
+import Paper from '@material-ui/core/Paper';
+
+
+function UserList({ users, click }) {
+
+    function createRow(user) {
+        return (
+            <UserItem key={user.id} id={user.id} userName={user.userName} age={user.age} click={click}></UserItem>
```
Comment (roeeyud) [inline src/components/UserList.js]:
```
click={() => click(user.id)}
```

File: src/components/UserListBuilder.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+    componentDidMount() {
+        const getUsers = async () => {
+            let res = await axios.get('https://aviehu-deployed-site.herokuapp.com/users');
+            return this.setState({ users: res.data })
+        }
+        getUsers();
+    }
+    render() {
+        if (this.state.users == null) {
```
Comment (roeeyud) [inline src/components/UserListBuilder.js]: const { users } = this.state

Comment (roeeyud) [reply src/components/UserListBuilder.js]: const users = this.state.users

Comment (roeeyud) [inline src/components/UserListBuilder.js]:
```js
user ? 
<ClipLoader
                    className={override}
                    sizeUnit={"px"}
                    size={200}
                    color={'#123abc'}
                    loading={this.state.loading}
                /> :
<div>
           <UserList users={this.state.users} click={this.clickHandler}></UserList>
           .... 
```

File: src/App.js [hunk 9]
```diff
@@ -1,22 +1,13 @@
...
     return (
-      <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <h1 className="App-title">Welcome to React</h1>
-        </header>
-        <p className="App-intro">
-          <code></code>
-        </p>
+      <div>
```
Comment (roeeyud) [inline src/App.js]: no need for div... 
In general, avoid using div unless absolutely necessary.
See react fragment documentation

File: src/components/UserForm.js [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import PropTypes from 'prop-types';
+
+function UserForm({ open }) {
+
+    return (
+        <div>
```
Comment (roeeyud) [inline src/components/UserForm.js]: not sure this div is needed

File: src/components/UserListBuilder.js (hunk 4 above)
Comment (roeeyud) [inline src/components/UserListBuilder.js]: Click handler still has the same issue...

**Pull request merged**